### PR TITLE
Add Experimental and Partner label

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Add an "Experimental" and "Partner" label to the Marketplace. Both will be used in the upcoming Marketplace revamp. 

Screenshot from 2020-11-20 03-37-47

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-30241
https://mattermost.atlassian.net/browse/MM-30811

### CODE REVIEW

File: internal/model/label.go [hunk 1]
```diff
@@ -9,16 +9,30 @@ type Label struct {
...
+}
 var CommunityLabel Label = Label{
 	Name:        "Community",
 	Description: "This plugin is maintained by the Open Source Community.",
 	URL:         "https://mattermost.com/pl/default-community-plugins",
 }
 
+var ExperimentalLabel Label = Label{
+	Name:        "Experimental",
+	Description: "This plugin is marked as experimental and not meant for production use. Use with caution!",
```
Comment (hanzei) [inline internal/model/label.go:32]: @aaronrothschild What do you think about this wording?

**Review by aaronrothschild — APPROVED** [review]
LGTM

**Review by larkox — APPROVED** [review]

File: internal/model/label.go [hunk 2]
```diff
@@ -26,6 +33,12 @@ var BetaLabel Label = Label{
 	URL:         "https://mattermost.com/pl/default-beta-plugins",
 }
 
+var ExperimentalLabel Label = Label{
+	Name:        "Experimental",
+	Description: "This plugin is marked as experimental and not meant for production use. Use with caution!",
```
Comment (mickmister) [inline internal/model/label.go:38]: I'm not sure what the tone of urgency should be but 0/5 this seems good to me

```suggestion
	Description: "This plugin is marked as experimental and not meant for production use. Please use with caution.",
```

**Review by mickmister — APPROVED** [review]

File: internal/model/label.go (hunk 2 above)
Comment (hanzei) [reply internal/model/label.go:38]: @aaronrothschild What do you think?

**Comment by hanzei** [comment]
@DHaussermann You can find the test steps in the two jira tickets. Let me know if you have any questions.

**Review by DHaussermann — COMMENTED** [review]
@hanzei after following the steps I do see the `Experimental` and `Partner` labels

That said When I run the Marketplace I don't see the labels for `Community`,`E20` and `Beta` are missing. I notice that several plugins such as Channel Recommender and MS Teams calling and Digital Ocean are missing. See screen.
Screen Shot 2021-01-28 at 5 01 23 PM
Interestingly I see this same behavior on the `master` branch.   

This may well be unrelated to the PR but, it prevents me from regression testing other labels. Any thoughts on what is occurring here?

**Comment by hanzei** [comment]
@DHaussermann Just to clarify here: The QA steps ensure the local marketplace in empty i.e. only the newly added plugin with the "Partner" label is there. Hence, it's expected to only see that plugin. The other plugin you see are the local installed ones.

I will make it more clear the next time what the expected behaviour is.

**Comment by DHaussermann** [comment]
Thanks @hanzei  sorry I did not realize I was looking at the combined local+prepackaged list. When I follow your steps on a clean server, I see modified plugin with the labels. Pre-packaged plugins are also showing which confused me but it would be expected that pre-packaged plugins also appear.      

Tested and passed
When following steps to clear marketplace and adding the labels:
- Labels display as expected
- Labels have tooltips and are styled as expected
- Reverted to stock cofig to pull all other plugins in and ensured there are no regression on existing `Community`,`E20` and `Beta` labels  

LGTM!

Created https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3815 to cover testing of new labels.

**Pull request merged**